### PR TITLE
Merged ww2ogg into the tool, added ffmpeg

### PR DESCRIPTION
**Pull request by pdemian**
1) Rewrote ww2ogg in C#. Used that instead of vgmsteam because we need to convert back into wem at some point and vgmsteam doesn't support that.
2) Added ffmpeg to convert ogg to other formats
3) Added video conversion framework, but it turns out that ffmpeg does not support bk2, only bik files.
4) Added new option --audioformat/-a

Known issues:
1) The files spit out by ww2ogg are fairly low quality and need to be revorb'd. Thinking of including the .exe or integrating the code into the tool
2) Some .wem files cannot be converted to formats other than ogg. Not sure if this is a decoding issue, or issue stated in #1

### CODE REVIEW

**Comment by avail** [comment]
ffmpeg is way too big to include in the repository. it should be an optional dependency, not required.

**Comment by pdemian** [comment]
> ffmpeg is way too big to include in the repository. it should be an optional dependency, not required.

Sure. I'll remove it and add instructions on how to install it.

**Comment by Traderain** [comment]
For creating wem files you should use Wwise it's free :). For modding existing files you only need to merge the WEM RIFF header and rebuild the soundbanks referencing the files. Both implemented already here: https://github.com/WolvenKit/Wolven-kit/tree/master/WolvenKit.Wwise

**Comment by pdemian** [comment]
> For creating wem files you should use Wwise it's free :). For modding existing files you only need to merge the WEM RIFF header and rebuild the soundbanks referencing the files. Both implemented already here: https://github.com/WolvenKit/Wolven-kit/tree/master/WolvenKit.Wwise

@Traderain So, what should happen to this pull request then?

**Comment by pdemian** [comment]
Didn't receive any information on how to move forward, so I'm closing this pull request for now.

**Comment by Traderain** [comment]
> Didn't receive any information on how to move forward, so I'm closing this pull request for now.

Sorry, I was a bit overwhelmed with work. I'll make a review.

File: CP77.Common/Audio/AudioconvWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        aac, 
+        flac 
+    }
+
+    public static class AudioconvWrapper
+    {
+
+        public static string Convert(string outDir, string filename, string filepath, EAudioExtension filetype)
+        {
+            string ffmpegpath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "avconv/ffmpeg.exe");
```
Comment (Traderain) [inline CP77.Common/Audio/AudioconvWrapper.cs:26]: Something like this would be better: https://ffmpeg.xabe.net/index.html

**Review by Traderain — COMMENTED** [review]
If you update the way ffmpeg is used we can merge it but I would rather urge you to update the existing code in wolvenkit in Wolvenkit.Wwise since this project is getting merged into that soon 🙂

**Pull request closed without merging**